cmd/tether/attach.go

```diff
@@ -37,18 +38,55 @@ const (
 var server attachServer
 
 type attachServer interface {
+	tether.Extension
+
 	start() error
 	stop()
 }
 
 // conn is held directly as it's how we stop the attach server
 type attachServerSSH struct {
-	conn   *net.Conn
-	config *ssh.ServerConfig
+	conn      *net.Conn
+	config    *tether.ExecutorConfig
+	sshConfig *ssh.ServerConfig
 
 	enabled bool
 }
 
+// Reload - tether.Extension implementation
+func (t *attachServerSSH) Reload(config *tether.ExecutorConfig) error {
+	defer trace.End(trace.Begin("attach reload"))
+
+	t.config = config
+	// process the sessions and launch if needed
+	for id, session := range config.Sessions {
+		log.Debugf("Processing config for session %s", id)
+		if session.Attach {
+			log.Debugf("Session %s is configured for attach", id)
+			// this will return nil if already running - calling server.start not t.start so that
+			// test impl gets invoked
+			err := server.start()
```

Comment:
`server` is a global for an `attachServer`, yet in the impl, we refer to the global.  How about just calling `t.start()`.